libraries/ESP8266WebServer/src/ESP8266WebServer.cpp

```diff
@@ -490,6 +490,11 @@ void ESP8266WebServer::_streamFileCore(const size_t fileSize, const String & fil
   send(200, contentType, "");
 }
 
+String ESP8266WebServer::pathArg(unsigned int i) { 
```

Comment:
Since you are returning an already allocated string, then return-type can be `const String&`.
That would save time and heap.
About `return ""` below, maybe we could define a global `extern String nullString;` in `WString.h`
@devyte what do you think ?

Reply:
No, that would uselessly allocate an empty String object from the heap.
I think the correct thing to do here is to return String().

Reply:
See my other comment about returning a const ref.

Reply:
I manually patched constref and used `emptyString`.